axis/models/event.py

```diff
@@ -122,12 +122,17 @@ def traverse(
 
 
 def extract_name_value(
-    data: dict[str, list[dict[str, str]] | dict[str, str]],
+    data: dict[str, list[dict[str, str]] | dict[str, str]], prefer: str | None = None
 ) -> tuple[str, str]:
     """Extract name and value from a simple item, take first dictionary if it is a list."""
     item = data.get("SimpleItem", {})
     if isinstance(item, list):
-        item = item[0]
+        if prefer is None:
+            item = item[0]
+        else:
+            item = next(
+                (item for item in item if item.get("@Name", "") == prefer), item[0]
+            )
```

Comment:
Sorry for not commenting earlier, I've been thinking if this would better keep the whole dict as the other information also could be interesting to expose. What do you think

Reply:
It might be interesting, but I didn't quite understand the interaction of all components and just wanted a minimally invasive fix for the bug.